scene.js

```diff
@@ -72,31 +73,41 @@ class Scene {
         mat4.multiply(this.viewMatrix, yRotation, xRotation);
         this.viewMatrix[14] = -translate;
 
-        var firstNode = this.nodes[0];
-
-        drawNodeRecursive(this, firstNode, mat4.create());
+        // draw the scene; note - do not assume a single fixed root
+        for (var ni = 0; ni < this.scene.nodes.length; ni++) {
+            var firstNode = this.nodes[this.scene.nodes[ni]];
+            drawNodeRecursive(this, firstNode, mat4.create());
+        }
 
         // draw to the front buffer
         this.frontBuffer.drawImage(this.backBuffer, 0, 0);
     }
 
-    loadImage(imageInfo, gl) {
+    loadImage(imageInfo, gl,mesh) {
```

Comment:
Yes, the old method could easily run out of samplers, that's a problem that I'm glad you're trying to fix.  I'm just concerned that the new code appears to make the assumption that each image is "owned" by only a single mesh.  I'll admit I didn't test it directly yet, it's just the impression I get from looking at these diffs.  It looks like the new code is tying each image to a single mesh, and that's an even older problem that I had (cheaply) fixed by going to the current method of one sampler per image.  We do have sample models where multiple meshes use the same image, and even multiple `gltf.texture` blocks reference the same `gltf.image` block with different sampler settings.  The "TextureSettingsTest" is the harshest of these, and does not yet work in the master branch.